core/src/main/scala/pureconfig/DerivedWriters.scala

```diff
@@ -108,8 +119,27 @@ trait DerivedWriters1 {
     }
   }
 
+  // Auxiliary type to make sure we always return `ConfigList` when writing HLists.
+  private[pureconfig] trait HListConfigWriter[T <: HList] {
```

Comment:
In this case, why didn't you define just `deriveHNilConfigWriter` and `deriveHConsConfigWriter` publicly? Since those are the "canonical" ways of writing HLists to configs and `deriveLabelledGenericInstance` asks specifically for a `WrappedConfigWriter`, we could simplify this. I think the same thing applies to readers, if you can get rid of `HListConfigReader`.